protocol/tlv/tlv.go

```diff
@@ -0,0 +1,73 @@
+package tlv
+
+import (
+	"encoding/binary"
+	"errors"
+	"fmt"
+	"io"
+
+	"github.com/iotaledger/hive.go/protocol/message"
+)
+
+var (
+	// ErrInvalidMessageLength is returned when a packet advertises a message length which
+	// is invalid for the given message type.
+	ErrInvalidMessageLength = errors.New("invalid message length")
+)
+
+const (
+	MessageTypeHeader message.Type = 0
+
+	// The amount of bytes dedicated for the message type in the packet header.
+	HeaderTypeBytesLength = 1
+
+	// The amount of bytes dedicated for the message length denotation in the packet header.
+	HeaderLengthByteLength = 2
+
+	// The amount of bytes making up the protocol TLV packet header.
+	HeaderBytesLength = HeaderLengthByteLength + HeaderTypeBytesLength
+)
+
+var (
+	// HeaderMessageDefinition is the message header sent in each message denoting the TLV fields.
+	HeaderMessageDefinition = &message.Definition{
+		ID:             MessageTypeHeader,
+		MaxBytesLength: HeaderBytesLength,
+		VariableLength: false,
+	}
+)
+
+// Header includes the definition of the message and its bytes length.
+type Header struct {
+	// The definition of the message.
+	Definition *message.Definition
+	// The length in bytes of the message.
+	MessageBytesLength uint16
+}
+
+// WriteHeader writes a TLV header into the given Writer.
+func WriteHeader(buf io.Writer, msgType message.Type, msgBytesLength uint16) error {
+	if err := binary.Write(buf, binary.BigEndian, byte(msgType)); err != nil {
```

Comment:
Switch to use `binary.LittleEndian` everywhere. Big endianess is a relic from when I rewrote the networking code in IRI and didn't check what endianess `BytesBuffer` used per default in Java. Given that most modern systems use little endianness, please use it.